app/backend/src/couchers/models.py

```diff
@@ -194,6 +185,12 @@ class User(Base):
     new_email_token_created = Column(DateTime(timezone=True), nullable=True)
     new_email_token_expiry = Column(DateTime(timezone=True), nullable=True)
 
+    # for verifying their phone number
+    phone_verification_token = Column(String(6), nullable=True)  # randomly generated Luhn 6-digit string
+    phone_verification_sent = Column(DateTime(timezone=True), nullable=True)
+    phone_verification_verified = Column(DateTime(timezone=True), nullable=True)
+    phone_verification_attempts = Column(Integer, nullable=False, default=0)
```

Comment:
Could you write a brief comment here for each as to when each is null and when not and what it's used for (e.g. `phone_verification_sent` is not reset when the phone is changed, it's kept to keep track of the last time a message was sent, etc)

Reply:
Ascii art state chart pushed! See if you can understand it and if you believe it is possible to make database constraints that checks that we are in one of the boxes. (or if you have any suggested simplification).

We could initialize times to 1970-01-01 instead of None and make them nullable=False.

Reply:
Yes, it makes sense... I need a moment to digest this :P

One thing that comes to mind is; I think `ChangePhone("")` should maybe be its own RPC (`ClearPhone`?) since it's completely decoupled?

Reply:
`ChangePhone("")` could even be removed completely. Even if I decide to be phoneless, why wouldn't I enjoy a higher verification score for 2 years for free? :D

Reply:
I think this is correct. Some small nitpicks:

* One missing thing is that `someone else verifies xx` should lead from any state to `Start`.
* There should maybe be some difference in the diagram between the variables in states `Code expired` and `Code sent` and similarly with `Verified` and `Verification expired`
* `Code sent` should lead to `Code sent` through `VerifyPhone(wrong)` as well

Reply:
I think it'd be good to add a db constraint to basically say `(verified is not null) => ((phone is not null) and (token is null) and (sent is not null))` and a `unique index on phone WHERE verified is not null`

Reply:
Okay, I checked the code and it seems correct. Could you add those db constraints, then I'm happy to approve this PR! (And also if you feel like tweaking the diagram, but not so important.)

Reply:
I managed the nonimportant stuff, but the constraints I failed to do. I don't think postgres has any UNIQUE .. WHERE construction.

Reply:
Also thanks for fixing up the diagram and other comments!

Reply:
Your logic was incorrect, (p => q) iff (~p OR q), but I got the idea how to write these things. I came up with a compact expression that checks many things at once. Thanks for the secret into using an index for unique-where constraints!